Modix.Services/Promotions/CreateCampaignContinuationData.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using Discord;
+using Modix.Data.Models.Core;
+
+namespace Modix.Services.Promotions
+{
+    public class CreateCampaignContinuationData
+    {
+        public CreateCampaignContinuationData(
+            IGuildUser subject, GuildRoleBrief targetRankRole, IEnumerable<GuildRoleBrief> rankRoles, string comment, ulong nominatingUserId)
+            => (Subject, TargetRankRole, RankRoles, Comment, NominatingUserId) = (subject, targetRankRole, rankRoles, comment, nominatingUserId);
+
+        public string Comment { get; }
+
+        public ulong NominatingUserId { get; }
+
+        public IEnumerable<GuildRoleBrief> RankRoles { get; }
+
+        public IGuildUser Subject { get; }
+
+        public GuildRoleBrief TargetRankRole { get; }
+
+        public void Deconstruct(
```

Comment:
Is this actually used anywhere? I certainly can't find it.

Reply:
I did a double take, too, as this is the first time I've actually seen this used. This method is how [this line](https://github.com/discord-csharp/MODiX/pull/196/files/99b8fc7c16848a412c4713800783ae875a4f3f40#diff-9bd04b98f5322decb6467f3552ae18c0R11) works. See [Deconstructing user-defined types](https://docs.microsoft.com/en-us/dotnet/csharp/deconstruct#deconstructing-user-defined-types).

Reply:
Ahh, gotcha. I was only looking for it in the service methods.

Yeah, I'm definitely not a fan of that syntax, over just a traditional constructor that sets all the properties.

Reply:
Haha, yeah, deconstructor is kinda weird. I made the class a POCO now, so no point in keeping the deconstructor anyway.